Implement functionality to fix invalid examples

StarKhan6368 opened the pull request:
<!--
Thanks for your interest in the project. Bugs filed and PRs submitted are appreciated!

Please make sure that you are familiar with and follow the Code of Conduct for
this project (found in the CODE_OF_CONDUCT.md file).

Also, please make sure you're familiar with and follow the instructions in the
contributing guidelines (found in the CONTRIBUTING.md file).

Please fill out the information below to expedite the review and (hopefully)
merge of your pull request!
-->

<!-- What changes are being made? (What feature/bug is being fixed here?) -->

**What**: Implement functionality to correct invalid examples.

<!-- How were these changes implemented? -->

**How**:

<!-- Have you done all of these things?  -->

**Checklist**:

<!-- add "N/A" to the end of each line that's irrelevant to your changes -->

<!-- to check an item, place an "x" in the box like so: "- [x] Documentation" -->

- [ ] Documentation added to the README.md OR link to PR on https://github.com/specmatic/specmatic-documentation
- [ ] Tests
- [ ] Sonar Quality Gate

<!-- feel free to add additional comments -->

StarKhan6368 committed:
[WIP] Add functionality to fix invalid examples

StarKhan6368 committed:
fixes the Fix button in GUI

yogeshnikam671 committed:
Add examples fix command

StarKhan6368 committed:
Fix disabling on fix-button in GUI

yogeshnikam671 committed:
Refactor examples fix command

yogeshnikam671 committed:
Add ability to fix schema examples

yogeshnikam671 committed:
Update examples fix command to show the errors and fix a bug with content type matching while finding out the scenario associated to a request

yogeshnikam671 committed:
Add ability to pass explicit examples directory name to fix examples command

joelrosario committed:
Merge branch 'main' into examples-fix-value

yogeshnikam671 committed:
Refactor fixExample methods in ExamplesInteractiveServer

StarKhan6368 committed:
Modify AnyPattern fixValue logic

joelrosario reviewed (changes requested):
- Can you introduce cycle prevention when fixing values
- Can you fix the feedback mechanism inside AnyPattern.getDiscriminatorPattern so that failures aren't turned into exceptions.

StarKhan6368 committed:
Logic to detect mandatory cyclic refs in ObjPattern

- Should throw an error when trying to fix a value where the pattern
  has unavoidable mandatory cyclic references.

StarKhan6368 committed:
Fix AnyPattern.getDiscriminatorPattern conversion

- Should not convert failures to exceptions as a way to signal back

StarKhan6368 committed:
Add tests for AnyPattern.fixValue

StarKhan6368 committed:
Add tests for HttpHeadersPattern.fixValue

- Fix dictionary-lookup not happening on fixValue

StarKhan6368 committed:
Add tests for HttpPathPattern.fixValue

- Fix dictionary-lookup not happening on fixValue

StarKhan6368 committed:
Add tests for HttpQueryParamPattern.fixValue

- Fix dictionary-lookup not happening on fixValue

StarKhan6368 committed:
Add extensible schema tests for JSONObjectPattern

StarKhan6368 committed:
Add and fix tests for ListPattern.fixValue

StarKhan6368 committed:
Fix cyclic checks use existing resolver code.

StarKhan6368 committed:
Simply fixValue method in Pattern

StarKhan6368 committed:
Modify Headers fixValue to reuse fix method.

StarKhan6368 committed:
Modify QueryParams fixValue to reuse fix method.

StarKhan6368 committed:
Remove unused code and simplify.

StarKhan6368 committed:
Fix flaky test in ListPatternTest

StarKhan6368 committed:
Headers and QueryParams are exempt from AllPatternsMandatory

StarKhan6368 committed:
QueryParamsPattern fix edge cases.

joelrosario committed:
Merge branch 'main' into examples-fix-value

joelrosario committed:
Added test for two-level unavoidable cyclic reference

joelrosario committed:
Merge branch 'main' into examples-fix-value

StarKhan6368 committed:
Fix getDiscriminatorBasedPattern when keys is optional
